local-proxy/src/server.ts

```diff
@@ -66,10 +66,20 @@ app.post('/invoke', json(), async (req, res) => {
   const { path, handler, args } = req.body;
   registry.register(requestId);
   try {
-    await transpilePromise;
-    // Make sure we use tmpDir as absolute path
-    const mod = require(pathJoin(tmpDir, path));
-    const fn = mod[handler];
+    let fn: (...args: any[]) => any;
+    if (path === '@binaris/shift-db' && (handler === 'getVersioned' || handler === 'poll')) {
+      fn = require(path)[handler];
+    } else {
+      await transpilePromise;
+      const joinedDir = pathResolve(tmpDir, path);
+      if (!joinedDir.startsWith(tmpDir)) {
+        return res.status(403).send({
+          error: 'Tried to reference path outside of root dir',
```

Comment:
```suggestion
          error: 'Cannot reference path outside of root dir',
```

Should we also say _which_ path it was, or will the backtrace be enough to show which file required which path?

Reply:
I'll add it.